go-controller/pkg/node/node_ip_handler_linux.go

```diff
@@ -0,0 +1,149 @@
+// +build linux
+
+package node
+
+import (
+	"net"
+	"sync"
+
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/kube"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/util"
+
+	"github.com/vishvananda/netlink"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/klog/v2"
+	utilnet "k8s.io/utils/net"
+)
+
+type addressManager struct {
+	addresses      sets.String
+	nodeAnnotator  kube.Annotator
+	mgmtPortConfig *managementPortConfig
+	mu             sync.Mutex
```

Comment:
@trozet I tend to just embed sync.Mutex in the struct and do `c.Lock()` but that's just a nit.